api/cpp/esp-idf/slint/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+# Copyright © SixtyFPS GmbH <info@slint.dev>
+# SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+idf_component_register(
+  SRCS "src/esp_slint.cpp"
+  INCLUDE_DIRS "include"
+  REQUIRES "esp_lcd" "esp_lcd_touch"
+  PRIV_REQUIRES "esp_timer")
+
+set(Rust_CARGO_TARGET "xtensa-${IDF_TARGET}-none-elf")
+set(CORROSION_NO_WARN_PARSE_TARGET_TRIPLE_FAILED ON)
+set(SLINT_FEATURE_STD OFF)
+set(SLINT_FEATURE_RENDERER_SOFTWARE ON)
+set(SLINT_TARGET_CARGO_FLAGS "-Zbuild-std=core,alloc")
+set(CMAKE_BUILD_TYPE Release)
+set(BUILD_SHARED_LIBS OFF)
+
+# FIXME: this never uses the current git repo
```

Comment:
There's a fix for that in the example by using override_path: https://docs.espressif.com/projects/idf-component-manager/en/latest/guides/packaging_components.html#adding-dependency-on-the-component-for-examples

Reply:
i don't think this helps.
I don't want to get Slint from FetchContent at all.

Reply:
If you don't want Slint from FetchContent, then how are you going to create the repo for the idf component registry?

I think that if you want one component and no fetch-content, you need to place the idf_component.yml into the top-level directory and change our top-level CMakeLists.txt to detect that it's being built as component and act as such. I don't see how it's possible otherwise, unless you create some extra packaging step.

Reply:
I want Slint from FetchContent when someone gets the component from the idf component registery.
And I want Slint from the current repository when building the example from the same repo.

Reply:
Hmm, I see. Maybe you could set a cmake (cache) variable when building the example? Like a `SLINT_ESP_IDF_PATH` or `SLINT_ESP_IDF_PATH_TO_REPO` and if that's set then maybe just use it with `add_subdirectory` instead of `FetchContent`?

When building the example, where do you invoke CMake? Top-level or in the example?